WTG.Analyzers/Analyzers/Array/ArrayCodeFixProvider.cs

```diff
@@ -0,0 +1,88 @@
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Simplification;
+
+namespace WTG.Analyzers
+{
+	[ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(ArrayCodeFixProvider))]
+	[Shared]
+	public sealed class ArrayCodeFixProvider : CodeFixProvider
+	{
+		public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(Rules.PreferArrayEmptyOverNewArrayConstructionDiagnosticID);
+
+		public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+		public override Task RegisterCodeFixesAsync(CodeFixContext context)
+		{
+			var diagnostic = context.Diagnostics.First();
+
+			switch (diagnostic.Id)
+			{
+				case Rules.PreferArrayEmptyOverNewArrayConstructionDiagnosticID:
+					context.RegisterCodeFix(
+						CodeAction.Create(
+							title: "Change to Array.Empty<T>()",
+							createChangedDocument: c => ReplaceWithArrayEmpty(context.Document, diagnostic, c),
+							equivalenceKey: "ChangeToArrayEmptyT"),
+						diagnostic: diagnostic);
+					break;
+			}
+
+			return Task.CompletedTask;
+		}
+
+		static async Task<Document> ReplaceWithArrayEmpty(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+		{
+			var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+			var diagnosticSpan = diagnostic.Location.SourceSpan;
+			var node = root.FindNode(diagnosticSpan);
+			var model = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+			if (node.Kind() == SyntaxKind.ArrayCreationExpression)
+			{
+				var syntax = (ArrayCreationExpressionSyntax)node;
+
+				var arrayType = syntax.Type.ElementType;
+				if (syntax.Type.RankSpecifiers.Count > 1)
+				{
+					var specifiers = syntax.Type.RankSpecifiers.RemoveAt(0);
+					arrayType = SyntaxFactory.ArrayType(arrayType).WithRankSpecifiers(specifiers);
+				}
+
+				document = document.WithSyntaxRoot(
+					root.ReplaceNode(
+						node,
+						CreateArrayEmptyInvocation(
+							arrayType)));
```

Comment:
Should probably copy the node's trivia onto it's replacement.
`CreateArrayEmptyInvocation(arrayType.WithoutTrivia()).WithTriviaFrom(node)`

so:

    int[] Magic() =>
        new int [0];
becomes:

    int[] Magic() =>
        Array.Empty<int>();

Reply:
I'll add another test case for that. 😄